File: Head_First_Design_Patterns/src/abstract_factory/python/PizzaTestDrive.py
# -*- coding: utf-8 -*-
"""
Created on Fri Oct 19 22:04:33 2018

@author: 凯风
"""

from Store import NYPizzaStore, ChicagoPizzaStore

if __name__ == "__main__":
    
    nyStore = NYPizzaStore()
    chicagoStore = ChicagoPizzaStore()
    
    pizza = nyStore.orderPizza("cheese")
    print("Order a " + str(pizza) + "\n\n")

    pizza = chicagoStore.orderPizza("veggie")
    print("Order a " + str(pizza) + "\n\n")        

File: Head_First_Design_Patterns/src/abstract_factory/PizzaTestDrive.java
package abstract_factory;

public class PizzaTestDrive {
    public static void  main (String[] args){
        PizzaStore nyStore = new NYPizzaStore();
        PizzaStore chicagoStore = new ChicagoPizzaStore();

        Pizza pizza = nyStore.orderPizza("cheese");
        System.out.println("Ordered a " + pizza + "\n\n");

        pizza = chicagoStore.orderPizza("clam");
        System.out.println("Ordered a " + pizza + "\n\n");
    }
}
